File: android/src/main/kotlin/com/shareinvest/flutter_kakao_maps/model/FadeInOutOptions.kt
package com.shareinvest.flutter_kakao_maps.model

import org.json.JSONObject

data class FadeInOutOptions(
    var fadeInTime: Int,
    var fadeOutTime: Int,
    var retentionTime: Int,
)


fun JSONObject.toFadeInOutOptions(): FadeInOutOptions {
    return FadeInOutOptions(
        this.getInt("fadeInTime"),
        this.getInt("fadeOutTime"),
        this.getInt("retentionTime"),
    )
}

File: lib/src/model/fade_in_out_options.dart
part of 'package:flutter_kakao_maps/flutter_kakao_maps.dart';

class FadeInOutOptions {
  final int fadeInTime;
  final int fadeOutTime;
  final int retentionTime;

  const FadeInOutOptions({
    required this.fadeInTime,
    required this.fadeOutTime,
    required this.retentionTime,
  });

  FadeInOutOptions copyWith({
    int? fadeInTime,
    int? fadeOutTime,
    int? retentionTime,
  }) =>
      FadeInOutOptions(
        fadeInTime: fadeInTime ?? this.fadeInTime,
        fadeOutTime: fadeOutTime ?? this.fadeOutTime,
        retentionTime: retentionTime ?? this.retentionTime,
      );
}

extension FadeInOutOptionsExtension on FadeInOutOptions {
  Map<String, dynamic> toMap() {
    return {
      "fadeInTime": fadeInTime,
      "fadeOutTime": fadeOutTime,
      "retentionTime": retentionTime,
    };
  }
}
